File: WarGamesAPI/Model/Question.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WarGamesAPI.Model;

public class Question
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public string? Text { get; set; }
    public DateTime? Date { get; set; }

    [ForeignKey("UserId")]
    public int UserId { get; set; }
    public User? User { get; set; }

    [ForeignKey("ConversationId")]
    public int ConversationId { get; set; }
    public Conversation? Conversation { get; set; }

    public ICollection<Answer> Answers { get; set; } = new List<Answer>();
} 

File: CodeRed/src/app/model/question/question.ts
export class Question {
    conversationId: number;
    text: string;
    mockReply: boolean;
    constructor(conversationId: number, text: string, mockReply: boolean) {
        this.conversationId = conversationId;
        this.text = text;
        this.mockReply = mockReply;
      }
}
